tests/mcproto/packets/status/test_status.py

```diff
@@ -1,67 +1,36 @@
-
-from mcproto.buffer import Buffer
 from mcproto.packets.status.status import StatusResponse
+from tests.helpers import gen_serializable_test
 
-
-@pytest.mark.parametrize(
-    ("data", "expected_bytes"),
-    [
+gen_serializable_test(
+    context=globals(),
+    cls=StatusResponse,
+    fields=[("data", Dict[str, Any])],
+    test_data=[
         (
-            {
-                "description": {"text": "A Minecraft Server"},
-                "players": {"max": 20, "online": 0},
-                "version": {"name": "1.18.1", "protocol": 757},
-            },
-            bytes.fromhex(
-                "797b226465736372697074696f6e223a7b2274657874223a2241204d696e6"
-                "5637261667420536572766572227d2c22706c6179657273223a7b226d6178"
-                "223a32302c226f6e6c696e65223a20307d2c2276657273696f6e223a7b226"
-                "e616d65223a22312e31382e31222c2270726f746f636f6c223a3735377d7d"
+            (
+                {
+                    "description": {"text": "A Minecraft Server"},
+                    "players": {"max": 20, "online": 0},
+                    "version": {"name": "1.18.1", "protocol": 757},
+                },
             ),
-        ),
-    ],
-)
-def test_serialize(data: dict[str, Any], expected_bytes: bytes):
-    """Test serialization of StatusResponse packet."""
-    expected_buffer = Buffer(expected_bytes)
-    # Clear the length before the actual JSON data. JSON strings are encoded using UTF (StatusResponse uses
-    # `write_utf`), so `write_utf` writes the length of the string as a varint before writing the string itself.
-    expected_buffer.read_varint()
-    expected_bytes = expected_buffer.flush()
-
-    buffer = StatusResponse(data=data).serialize()
-    buffer.read_varint()  # Ditto
-    out = buffer.flush()
-
-    assert json.loads(out) == json.loads(expected_bytes)
-
-
-@pytest.mark.parametrize(
-    ("read_bytes", "expected_data"),
-    [
-        (
             bytes.fromhex(
-                "797b226465736372697074696f6e223a7b2274657874223a2241204d696e6"
-                "5637261667420536572766572227d2c22706c6179657273223a7b226d6178"
-                "223a32302c226f6e6c696e65223a20307d2c2276657273696f6e223a7b226"
-                "e616d65223a22312e31382e31222c2270726f746f636f6c223a3735377d7d"
+                "84017b226465736372697074696f6e223a207b2274657874223a202241204"
+                "d696e65637261667420536572766572227d2c2022706c6179657273223a20"
+                "7b226d6178223a2032302c20226f6e6c696e65223a20307d2c20227665727"
+                "3696f6e223a207b226e616d65223a2022312e31382e31222c202270726f74"
+                "6f636f6c223a203735377d7d"
+                # Contains spaces that are not present in the expected bytes.
+                # "5637261667420536572766572227d2c22706c6179657273223a7b226d6178"
+                # "223a32302c226f6e6c696e65223a20307d2c2276657273696f6e223a7b226"
+                # "e616d65223a22312e31382e31222c2270726f746f636f6c223a3735377d7d"
```

Comment:
I don't follow, what do you mean? How come the data here are different from those in the test originally?

Reply:
This : 
```
{"description": {"text": "A Minecraft Server"}, "players": {"max": 20, "online": 0}, "version": {"name": "1.18.1", "protocol": 757}
{"description":{"text":"A Minecraft Server"},"players":{"max":20,"online": 0},"version":{"name":"1.18.1","protocol":757}}
```
I ended up using 
```python
json.dumps(self.data, separators=(",", ":"))
```
to remove the useless spaces in the json output

Reply:
Did you? Where? I might just be blind, but I can't find that, `status.py` still just has `json.dump(self.data)` on line 50 (and so does `chat.py` on line 71).

Reply:
In a commit I didn't push yet sorry

Reply:
I pushed it